pkg/controller/master/image/vm_image_controller.go

```diff
@@ -57,6 +57,26 @@ func (h *vmImageHandler) OnChanged(_ string, image *harvesterv1.VirtualMachineIm
 			return h.images.Update(toUpdate)
 		}
 
+		// sync virtualSize (handles the case for existing images that were
+		// imported before this field was added, because adding the field to
+		// the CRD triggers vmImageHandler.OnChanged)
+		if image.Status.VirtualSize == 0 {
+			bi, err := util.GetBackingImage(h.backingImageCache, image)
+			if err != nil {
+				if !errors.IsNotFound(err) {
+					logrus.WithError(err).WithFields(logrus.Fields{
+						"namespace": image.Namespace,
+						"name":      image.Name,
+					}).Error("failed to get backing image for vmimage")
+					return image, err
+				}
+				return h.handleRetry(image)
+			}
+			toUpdate := image.DeepCopy()
+			toUpdate.Status.VirtualSize = bi.Status.VirtualSize
+			return h.images.Update(toUpdate)
+		}
```

Comment:
Just a nit, I prefer the `nest-if` with lower depth (do not have functional change)
```
		if image.Status.VirtualSize == 0 {
			bi, err := util.GetBackingImage(h.backingImageCache, image)
			if err == nil {
				toUpdate := image.DeepCopy()
				toUpdate.Status.VirtualSize = bi.Status.VirtualSize
				return h.images.Update(toUpdate)
			}

			if !errors.IsNotFound(err) {
				logrus.WithError(err).WithFields(logrus.Fields{
					"namespace": image.Namespace,
					"name":      image.Name,
				}).Error("failed to get backing image for vmimage")
				return image, err
			}

			return h.handleRetry(image)
		}
```

Reply:
code structure has changed with latest update